Utils/IntrinsicallySizedCollectionView.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  IntrinsicallySizedCollectionView.swift
+//  Movies
+//
+//  Created by Gabriel Rodrigues Minucci on 29/06/2022.
+//
+
+import UIKit
+
+class IntrinsicallySizedCollectionView: UICollectionView {
+    
+    override var contentSize: CGSize {
+        didSet {
+            invalidateIntrinsicContentSize()
+        }
+    }
+    
+    override func reloadData() {
+        super.reloadData()
+        invalidateIntrinsicContentSize()
+    }
+    
+    override var intrinsicContentSize: CGSize {
+        setNeedsLayout()
+        layoutIfNeeded()
+        return contentSize
+    }
```

Comment:
I've seen this solution as I was searching for an answer, but I didn't understand it. Would you explain it to me during our next call, please?

Reply:
For sure :) 
Please remind me in case I forget